activestorage/app/models/active_storage/blob.rb

```diff
@@ -155,6 +155,11 @@ def signed_id(purpose: :blob_id)
     super
   end
 
+  # Returns a signed ID for this blob that's suitable for direct uploads.
+  def direct_upload_signed_id(purpose: :direct_upload)
+    signed_id(purpose: purpose)
+  end
+
```

Comment:
Let’s get rid of this method and use `signed_id(purpose: :direct_upload)` in `DirectUploadsController`. I think that’ll look like this:

```ruby
def direct_upload_json(blob)
  {
    signed_id: blob.signed_id(purpose: :direct_upload),
    direct_upload: {
        url: blob.service_url_for_direct_upload,
        headers: blob.service_headers_for_direct_upload
    }
  }
end
```

Reply:
Added a commit addressing this comment. Notice that now we won't be sending the following attributes anymore: `id`, `filename`, `byte_size`, `checksum`, `metadata` and `content_type`. Not sure if there was a reason to send them in the past..

Reply:
Ah, sorry, I forgot that `as_json` with the `:methods` option includes attributes (and not just the provided methods). We can do this instead:

```ruby
def direct_upload_json(blob)
  blob.as_json(root: false).merge(
    signed_id: blob.signed_id(purpose: :direct_upload),
    direct_upload: {
      # ...
    }
  )
end
```